lldb/include/lldb/Core/ValueObject.h

```diff
@@ -668,6 +699,32 @@ class ValueObject {
   CreateValueObjectFromData(llvm::StringRef name, const DataExtractor &data,
                             const ExecutionContext &exe_ctx, CompilerType type);
 
+  static lldb::ValueObjectSP
+  CreateValueObjectFromBytes(lldb::TargetSP target_sp, const void *bytes,
+                             CompilerType type);
+
+  static lldb::ValueObjectSP CreateValueObjectFromBytes(lldb::TargetSP target,
```

Comment:
Do we need to be able to add a name for the value object? Some the creation APIs allow users to specify names and the native synthetic child providers could use these APIs if they can specify a name.